google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/EnhancedBigtableStubSettings.java

```diff
@@ -613,7 +618,7 @@ UnaryCallSettings<PingAndWarmRequest, Void> pingAndWarmSettings() {
 
   /** Returns a builder containing all the values of this settings class. */
   public Builder toBuilder() {
-    return new Builder(this);
+    return new Builder(this).setEndpoint(getEndpoint());
```

Comment:
`patchCredentials` takes in the StubSettings.Builder and not the actual StubSettings class: https://github.com/googleapis/java-bigtable/blob/26d5437903d5a611f079bb659ca5924227f0aae9/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/EnhancedBigtableStub.java#L234-L240

Implementation here: https://github.com/googleapis/java-bigtable/blob/26d5437903d5a611f079bb659ca5924227f0aae9/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/EnhancedBigtableStub.java#L400-L404 ends up calling the `getEndpoint()` from the Builder which returns the clientSettingsEndpoint and not the fully resolved endpoint.

This call is will set the resolved endpoint as the clientSettingsEndpoint.

Reply:
The primary reason it needs a builder is so that it can mutate the credentials in the builder.

I dont love the toBuilder() mutating the endpoint. How would you feel about changing signature of patchCredentials to take the endpoint and JwtAudienceMapping as parameters?

ie in EnhancedStub#createClientContext():

```patchCredentials(builder, settings.getEndpoint(), settings.getJwtAudienceMapping());```

Reply:
I think my original comment was prior to some fixes we made regarding how the Builder's `getEndpoint()` and actual class' `getEndpoint()` returned different results. They now return the same value.

I think it can remain a builder.